### ISSUE TITLE
Since Quarkus 3.14.3: `LogManager error of type GENERIC_FAILURE: Unable to find named filter 'otel-datadog-id-converter'`

### ISSUE THREAD

**Issue by cthiebault** [quarkusio/quarkus#43270]
### Describe the bug

Since Quarkus 3.14.3, the following error is thrown when starting the application:
```text
LogManager error of type GENERIC_FAILURE: Unable to find named filter 'otel-datadog-id-converter'
```

I have a custom LoggingFilter as follows:

```kotlin
@LoggingFilter(name = "otel-datadog-id-converter")
class OpenTelemetryDataDogIdConversionFilter(
  @ConfigProperty(name = "otel-datadog-id-converter.trace-id-key", defaultValue = "traceId")
  private val traceIdKey: String,
  @ConfigProperty(name = "otel-datadog-id-converter.span-id-key", defaultValue = "spanId")
  private val spanIdKey: String,
  @ConfigProperty(name = "otel-datadog-id-converter.parent-id-key", defaultValue = "parentId")
  private val parentIdKey: String
) : Filter {
  ...
}
```

It was working fine with Quarkus 3.14.2.

I don't know if it could be related, but I also had an issue with a custom Liquibase changeset: I get a `ClassNotFoundException`. Liquibase cannot find anymore my custom changeset class...

### Output of `java -version`

java 21.0.2, Kotlin 2.0.20

### Quarkus version or git rev

3.14.3

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Gradle 8.10

**Comment by geoand** [comment]
Any chance you can attach a sample?

Also, do you know if this only happens with Kotlin or does it affect Java as well?

**Comment by cthiebault** [comment]
I will try to attach a reproducer next week. I only user Gradle/Kotlin... I don't know for Java/Maven...

**Comment by geoand** [comment]
Thanks!

I am interested in the Gradle/Java combination in additiona to Gradle/Kotlin, if possible

**Comment by cthiebault** [comment]
I think it was the same issue as https://github.com/quarkusio/quarkus/issues/43242

It is now fixed with Quarkus 3.14.4!

**Comment by geoand** [comment]
Gotcha, thanks!
